addons/mass_mailing/models/mailing.py

```diff
@@ -1306,36 +1315,117 @@ def _unsubscribe_token(self, res_id, email):
         token = (self.env.cr.dbname, self.id, int(res_id), tools.ustr(email))
         return hmac.new(secret.encode('utf-8'), repr(token).encode('utf-8'), hashlib.sha512).hexdigest()
 
-    def _convert_inline_images_to_urls(self, body_html):
-        """
-        Find inline base64 encoded images, make an attachement out of
-        them and replace the inline image with an url to the attachement.
-        """
 
-        def _image_to_url(b64image: bytes):
-            """Store an image in an attachement and returns an url"""
-            attachment = self.env['ir.attachment'].create({
-                'datas': b64image,
-                'name': "cropped_image_mailing_{}".format(self.id),
-                'type': 'binary',})
+    def _image_to_url(self, b64image: bytes):
+        """Store an image in an attachement and returns an url"""
+        attachment = self.env['ir.attachment'].create({
+            'datas': b64image,
+            'name': "cropped_image_mailing_{}".format(self.id),
+            'type': 'binary',})
+
+        attachment.generate_access_token()
 
-            attachment.generate_access_token()
+        return '/web/image/%s?access_token=%s' % (
```

Comment:
While we are at it, could define a method "_compute_image_url" on "ir.attachment" model that does it (always better to encapsulate that to hide a bit details). Could be added probably directly in base module as the controller exists in base.